pilot/pkg/model/context.go

```diff
@@ -158,6 +166,49 @@ type Proxy struct {
 
 	// GlobalUnicastIP stores the globacl unicast IP if available, otherwise nil
 	GlobalUnicastIP string
+
+	// Generator is used to generate resources for the node, based on the PushContext.
+	Generator Generator
```

Comment:
Why made Generator per proxy, imo, we can put it in a common place.

Reply:
A proxy - based on version, interception mode, etc - can use a different generator. 

For example may use XDS v3 - or UDPA, or gRPC.

It's also per resource - it is more flexible.

Reply:
can you please call this gRPC generator? As flexible as it sounds to have different generators for different proxies based on people's preferences and such, the reality is that there will be only one config gen in Istio for our APIs. This entire thing is for isolating the impact of grpc. So from that point of view, it will be helpful to have the fields scoped to just gRPC so that others don't end up using it for envoy code.

Reply:
This is not intended only for gRPC generator - but as a replacement for the old Generator interface which is tied to v2 API and one-method per TypeUrl. 

Happy to convert the networking/core to use the new interface - after this PR is approved. 

The current interface is clearly not adequate - and I think what is proposed in this PR is a clear improvement.

Reply:
Shriram, to clarify: gRPC was just meant as an example, and a way to drive a much-needed cleanup and refactoring. My real goal is to move towards UDPA and more flexible use of XDS protocol - beyond just generating the 3 v2 types.